File: Java/Search/BinarySearch.java
package Java.Search;

class BinarySearch {
    static int search(int[] arr, int target, int low, int mid, int high, int size) {
        for (int i = 0; i < size; i++) {
            if (arr[mid] == target) {
                return mid;
            } else if (arr[mid] > target) {
                high = mid;
                mid = low + (high - low) / 2;
            } else {
                low = mid;
                mid = low + (high - low) / 2;
            }
        }
        return -1;
    }

    public static void main(String[] args){
        int[] arr = {2, 3, 5, 7, 9};
        int target = 3;
        int size = arr.length;

        int low = 0;
        int high = size - 1;
        int mid = low + (high - low) / 2;
        int result = search(arr, target, low, mid, high, size);

        if(result != -1){
            System.out.println("Number found at index of: " + result);
        } else{
            System.out.println("Number Not found");
        }
    }

}

File: Python/Search/BinarySearch.py
def binarySearch(arr, target):
    low = 0
    high = len(arr) - 1

    mid = low + (high - low) // 2

    for i in range(0, len(arr)):
        if arr[mid] == target:
            return mid
        elif arr[mid] > target:
            high = mid
            mid = low + (high - low) // 2
        else:
            low = mid
            mid = low + (high - low) // 2
    return -1

if __name__ == "__main__": 
    arr = [2, 3, 4, 10, 40] 
    x = 1
  
    result = binarySearch(arr, x) 
    if(result == -1): 
        print("Element is not present in array") 
    else: 
        print("Element is present at index", result) 